### PR TITLE
feat(validation): validate docx template on upload

### PR DESCRIPTION
**Pull request by winged**
Docx templates can contain syntax errors that will only get caught when
rendering the template.  This causes problems, as the user seeing the
problem (some frontend user for example) may not be the same person who
caused the problem (the person creating the template).

Thus we now validate the syntax on upload already.

### CODE REVIEW

**Comment by winged** [comment]
Note: this does not yet validate against the proposed placeholder list. I'll implement that in a follow-up PR instead

File: document_merge_service/api/engines.py [hunk 1]
```diff
@@ -35,6 +56,20 @@ class DocxMailmergeEngine(DocxValidator):
     def __init__(self, template):
         self.template = template
 
+    def _has_placeholders(self, part):
+        return bool(part.findall("//MergeField"))
+
+    def validate_template_syntax(self):
+        document = MailMerge(self.template)
+        # syntax can't be invalid as it's validated by office
+        # suites. However we need to have *some* placeholders
```
Comment (open-dynaMIX) [inline document_merge_service/api/engines.py]: Do we really must have placeholders? I know, it doesn't make sense to use MailMerge without them. But do we know it's never used that way so far?

Comment (winged) [reply document_merge_service/api/engines.py]: I assume that if you want to use the functionality, you also intend to have some placeholders. We could also leave out validation for this type of templates, but this way, we could also detect possible mismatch of the engine type.

Comment (czosel) [reply document_merge_service/api/engines.py]: How is the syntax validated by office suites?

**Pull request merged**